### PR TITLE
Use fedcloud secrets

### PR DESCRIPTION
**Pull request by enolfc**
# Summary

Change the approach for managing records, avoiding sending the long lived ones in cloud-init. At the same time move the ansible role to this repository instead of relying on a external role (EGI-Federation/ansible-role-fedcloud-ops) that makes no sense as an independent thing.

---

**Related issue :** 341

### CODE REVIEW

**Comment by enolfc** [comment]
Lots of testing happened here: #344

**Review by gwarf — APPROVED** [review]
LGTM (but mostly blind/direct approval to not block things, maybe it would be good to have some more people as reviewers of this)

File: deploy/roles/catchall/defaults/main.yaml [hunk 1]
```diff
@@ -0,0 +1,22 @@
+# AMS details
+ams_project: egi_cloud_info
+ams_host: msg.argo.grnet.gr
+ams_token: secret
+
+# check-in endpoint
+checkin_token_endpoint: "https://aai.egi.eu/oidc/token"
```
Comment (sebastian-luna-valero) [inline deploy/roles/catchall/defaults/main.yaml:7]: Legacy config?

**Review by sebastian-luna-valero — COMMENTED** [review]
Sorry, just going through this PR now...

File: .github/workflows/deploy.yml [hunk 2]
```diff
@@ -25,17 +25,17 @@ jobs:
...
           pip install yq git+https://github.com/tdviet/fedcloudclient.git
-          curl -L https://github.com/oidc-mytoken/client/releases/download/v0.3.0/mytoken_0.3.0_Linux_x86_64.tar.gz \
-            | tar -xzf -
-          mkdir ~/.mytoken
-          curl https://raw.githubusercontent.com/oidc-mytoken/client/master/config/example-config.yaml > ~/.mytoken/config.yaml
       - name: Configure providers access
         env:
-          MYTOKEN: ${{ secrets.MYTOKEN }}
           REFRESH_TOKEN: ${{ secrets.REFRESH_TOKEN }}
+          ANSIBLE_SECRETS: ${{ secrets.ANSIBLE_SECRETS }}
```
Comment (sebastian-luna-valero) [inline .github/workflows/deploy.yml:31]: Also, I guess these secrets should be directly stored in https://secrets.egi.eu/ ?

**Review by sebastian-luna-valero — COMMENTED** [review]
Sorry, an additional comment

File: .github/workflows/deploy.yml (hunk 2 above)
Comment (enolfc) [reply .github/workflows/deploy.yml:31]: yes, although I don't know yet what's the best way of dealing with this, any ideas?

Comment (sebastian-luna-valero) [reply .github/workflows/deploy.yml:31]: following up internally :)

**Pull request merged**